pkg/services/ngalert/store/deltas.go

```diff
@@ -11,12 +12,29 @@ import (
 // AlertRuleFieldsToIgnoreInDiff contains fields that are ignored when calculating the RuleDelta.Diff.
 var AlertRuleFieldsToIgnoreInDiff = [...]string{"ID", "Version", "Updated", "UpdatedBy"}
 
+// AlertRuleFieldsWhichAffectQuery contains fields which affect the rule's query(s)
+var AlertRuleFieldsWhichAffectQuery = [...]string{"Data", "Condition"}
```

Comment:
Based on team discussion, the `Condition` field doesn't meaningfully affect the underlying query and can be excluded from permission checks. We're aligning updates requiring query permission to changes in the `Data` struct (and likely `Interval`), which better reflects how query access is used elsewhere like in dashboards.

Reply:
Done 👍